src/DBFTPlugin/Consensus/ConsensusContext.MakePayload.cs

```diff
@@ -96,25 +97,38 @@ internal void EnsureMaxBlockLimitation(IEnumerable<Transaction> txs)
                 if (blockSystemFee > dbftSettings.MaxBlockSystemFee) break;
 
                 hashes.Add(tx.Hash);
-                Transactions.Add(tx.Hash, tx);
-                VerificationContext.AddTransaction(tx);
+                Transactions[pID].Add(tx.Hash, tx);
+                VerificationContext[pID].AddTransaction(tx);
             }
 
-            TransactionHashes = hashes.ToArray();
+            TransactionHashes[pID] = hashes.ToArray();
         }
 
-        public ExtensiblePayload MakePrepareRequest()
+        internal IEnumerable<Transaction> PickTransactions()
```

Comment:
@ZhangTao1596, maybe we can push this nomenclature and refactor change in another PR and then merge here.

Reply:
But it looks good here and fits more for this PR